SEUNAGBEYE/React-Redux-Es6
# #161367575 Save Course

### What does this PR do?
Create the course list table component for displaying the list of courses in the application

### Description of Task to be completed?
- create course form component
- map authors data for dropdown
- add author's reducer
- create initial state
- add create course functionalities
- add update course functionalities

### How should this be manually tested?
- fetch this branch: `git fetch origin ft-save-course-161367575`
- install all dependencies with `npm i`
- start the app `npm start -s`
- click on `Add Course` button to add a course
- fill in the course information and click on save
- then you will be redirected to the courses page where you will see the newly created course
- to edit a course, click on the course title
- you will be redirected to the course form with pre-populated data
- edit the data  and click on save

### Any background context you want to provide?
NA


### What are the relevant pivotal tracker stories?
[#161367575](https://www.pivotaltracker.com/n/projects/2200608/stories/161367575)

### Screenshots (if appropriate)
<img width="1229" alt="screen shot 2018-10-21 at 6 41 58 pm" src="https://user-images.githubusercontent.com/25517388/47272765-251e2f00-d582-11e8-8443-e4057d089158.png">
<img width="1053" alt="screen shot 2018-10-21 at 10 38 11 pm" src="https://user-images.githubusercontent.com/25517388/47272769-323b1e00-d582-11e8-9905-53cfb47ce040.png">
<img width="1045" alt="screen shot 2018-10-21 at 10 38 25 pm" src="https://user-images.githubusercontent.com/25517388/47272773-40893a00-d582-11e8-863f-f0d89db982b1.png">
